### PR TITLE
Manage categories with FireStore (uploading data)

### PR DESCRIPTION
**Pull request by ramshty**
# Implement Firestore Integration for Categories

## Description

This pull request introduces Firestore integration to manage categories and subcategories in the QuickFixApp.

### Changes:

- **Python Script**: Created `upload_categories.py` to upload all categories and subcategories into Firestore.
- **Data Classes**: Defined `Category` and `Subcategory` data classes in Kotlin.

### Goal
Avoid storing categories locally by fetching them from Firestore for better maintainability.

### Next Steps (in PR #)

- **Repository**: Implement a repository to fetch categories from Firestore.
- **ViewModel**: Set up a ViewModel to manage UI data.
- **Testing**: Write unit and integration tests.

Please review the current implementation. Further updates

### CODE REVIEW

File: upload_categories.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            }
+            subcategory_ref.set(subcategory_data)
+
+    print("Data uploaded successfully.")
+
+def main():
+    db = initialize_firestore()
+
+    # Define your categories, subcategories, and tags
+    categories = [
```
Comment (JitterKiller) [inline upload_categories.py:50]: Great initiative! Now, everyone can upload their new categories quickly without spending hours doing it manually. Fantastic job!

**Review by JitterKiller — APPROVED** [review]
LGTM! This will save so much time and streamline the process for everyone. Great work simplifying category uploads!

File: upload_categories.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        # Upload subcategories
+        for subcategory in category.subcategories:
+            subcategory_ref = category_ref.collection('subcategories').document(subcategory.id)
+            subcategory_data = {
+                'name': subcategory.name,
+                'tags': subcategory.tags
+            }
+            subcategory_ref.set(subcategory_data)
+
+    print("Data uploaded successfully.")
```
Comment (mohagpt) [inline upload_categories.py:44]: Just a small suggestion : consider adding validation for the data before uploading it to Firestore to ensure no empty fields or duplicate IDs.

**Review by mohagpt — APPROVED** [review]
This is a great PR! The addition of Category and Subcategory models, along with the script to upload them to Firestore, is well-structured and comprehensive. This implementation is a significant improvement as it enables the entire team to easily add and manage categories without much hassle. The predefined categories and subcategories offer a great starting point, and the extensibility of the upload script will make future updates seamless.

File: upload_categories.py (hunk 2 above)
Comment (ramshty) [reply upload_categories.py:44]: Ok I will modify the script accordingly.

**Pull request merged**